GTG/gtk/ui/general_preferences.ui

```diff
@@ -0,0 +1,390 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <object class="GtkScrolledWindow" id="general_pref_window">
+    <property name="min-content-height">500</property>
```

Comment:
this is set so that the general settings page (which is not so huge after all) fits nicely into the preferences window. As for the homogeneous size of stack children, this is secured automatically by GTK:
https://developer.gnome.org/gtk3/stable/GtkStack.html#GtkStack--vhomogeneous